modules/menu/menu.c

```diff
@@ -580,18 +582,77 @@ struct menu *menu_get(void)
 }
 
 
-void menu_uacur_set(struct ua *ua)
+struct call *menu_find_call(const char *id)
 {
-	menu.ua_cur = ua;
+	struct le *le;
+	struct call *call;
+
+	if (!str_isset(id))
+		return NULL;
+
+	for (le = list_head(uag_list()); le; le = le->next) {
+		struct ua *ua = le->data;
+		struct list *calls = ua_calls(ua);
+
+		call = call_find_id(calls, id);
+	}
+
+	return call;
 }
 
 
-struct ua *menu_uacur(void)
+/**
+ * Select active call
+ *
+ * @param id  The call-id
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int menu_selcall(struct re_printf *pf, const char *id)
 {
-	if (!menu.ua_cur)
-		menu.ua_cur = list_ledata(list_head(uag_list()));
+	struct call *call = menu_find_call(id);
+	struct list *calls = ua_calls(call_get_ua(call));
+
+	if (call) {
+		menu.callid = id;
```

Comment:
please note that the call-id string is dynamically allocated, and should be
copied using e.g. `str_dup`.

otherwise if the call object is destructed, the callid string will point to freed memory